pkg/admission/clusterworkspace/admission.go

```diff
@@ -126,3 +130,34 @@ func (o *clusterWorkspace) Validate(ctx context.Context, a admission.Attributes,
 
 	return nil
 }
+
+func (o *clusterWorkspace) Admit(ctx context.Context, a admission.Attributes, _ admission.ObjectInterfaces) (err error) {
+	if a.GetResource().GroupResource() != tenancyv1alpha1.Resource("clusterworkspaces") {
+		return nil
+	}
+
+	u, ok := a.GetObject().(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("unexpected type %T", a.GetObject())
+	}
+	cw := &tenancyv1alpha1.ClusterWorkspace{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, cw); err != nil {
+		return fmt.Errorf("failed to convert unstructured to ClusterWorkspace: %w", err)
+	}
+
+	if cw.Labels == nil {
+		cw.Labels = map[string]string{}
+	}
+	cw.Labels[tenancyv1alpha1.ClusterWorkspacePhaseLabel] = string(cw.Status.Phase)
+
+	for _, initializer := range cw.Status.Initializers {
+		cw.Labels[fmt.Sprintf("%s.%s", tenancyv1alpha1.ClusterWorkspaceInitializerLabelPrefix, initializer)] = ""
+	}
```

Comment:
and remove those that are gone. Did I mention this is complexity without value to care about the initializers in filtering? :)